omegaconf/base.py

```diff
@@ -403,30 +398,65 @@ def _resolve_interpolation(
             try:
                 # NB: `match_list` is ignored here as complex interpolations may be
                 # nested, the string will be re-parsed within `_evaluate_complex()`.
-                new = self._evaluate_complex(
+                return self._evaluate_complex(
                     value,
                     key=key,
                     throw_on_missing=throw_on_missing,
                     throw_on_resolution_failure=throw_on_resolution_failure,
                 )
-            except ParseError:
+            except ParseError as e:
+                if throw_on_resolution_failure:
+                    self._format_and_raise(key=key, value=value, cause=e)
                 return None
-            return StringNode(value=new, key=key)
         else:
             assert False
 
-    def _evaluate_simple(self, value: str, **kw: Any) -> Optional["Node"]:
+    def _evaluate_simple(
+        self,
+        value: str,
+        key: Any,
+        throw_on_missing: bool,
+        throw_on_resolution_failure: bool,
+    ) -> Optional["Node"]:
         """Evaluate a simple interpolation"""
         value_kind, match_list = get_value_kind(value=value, return_match_list=True)
         assert value_kind == ValueKind.INTERPOLATION and len(match_list) == 1
         match = match_list[0]
         node = self._resolve_simple_interpolation(
-            inter_type=match.group(1), inter_key=match.group(2), **kw,
+            inter_type=match.group(1),
+            inter_key=match.group(2),
+            key=key,
+            throw_on_missing=throw_on_missing,
+            throw_on_resolution_failure=throw_on_resolution_failure,
         )
         return node
 
-    def _evaluate_complex(self, value: str, **kw: Any) -> Optional[str]:
-        """Evaluate a complex interpolation (>1, nested, with other strings...)"""
+    def _evaluate_complex(
+        self,
+        value: str,
+        key: Any,
+        throw_on_missing: bool,
+        throw_on_resolution_failure: bool,
+    ) -> Optional["Node"]:
+        """
+        Evaluate a complex interpolation (>1, nested, with other strings...).
```

Comment:
add a few examples for complex interpolations here.

Reply:
Done in 46f6d60